spring-cloud-config-client/src/main/java/org/springframework/cloud/config/client/ConfigServicePropertySourceLocator.java

```diff
@@ -191,62 +193,47 @@ private RestTemplate getSecureRestTemplate(ConfigClientProperties client) {
 		SimpleClientHttpRequestFactory requestFactory = new SimpleClientHttpRequestFactory();
 		requestFactory.setReadTimeout((60 * 1000 * 3) + 5000); //TODO 3m5s, make configurable?
 		RestTemplate template = new RestTemplate(requestFactory);
+		String username = client.getUsername();
 		String password = client.getPassword();
 		String authorization = client.getAuthorization();
+		Map<String, String> headers = client.getHeaders();
```

Comment:
I would create a new Map here and copy values. There are no guarantees that the map is mutable.